lib/dry/core/basic_object.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+# This implementation was imported from `hanami-utils` gem.
+module Dry
+  module Core
+    # BasicObject
+    #
+    # @since 0.7.0
+    class BasicObject < ::BasicObject
+      # Lookups constants at the top-level namespace, if they are missing in the
+      # current context.
+      #
+      # @param name [Symbol] the constant name
+      #
+      # @return [Object, Module] the constant
+      #
+      # @raise [NameError] if the constant cannot be found
+      #
+      # @since 0.7.0
+      # @api private
+      #
+      # @see https://ruby-doc.org/core/Module.html#method-i-const_missing
+      def self.const_missing(name)
+        ::Object.const_get(name)
+      end
+
+      # Returns the class for debugging purposes.
+      #
+      # @since 0.7.0
+      #
+      # @see http://ruby-doc.org/core/Object.html#method-i-class
+      def class
+        (class << self; self; end).superclass
+      end
+
+      # Bare minimum inspect for debugging purposes.
+      #
+      # @return [String] the inspect string
+      #
+      # @since 0.7.0
+      #
+      # @see http://ruby-doc.org/core/Object.html#method-i-inspect
+      def inspect
+        "#<#{self.class}:#{"0x0000%x" % (__id__ << 1)}#{__inspect}>"
+      end
```

Comment:
it won't work on ruby 2.7, object_id there is not linked to memory location. It'd be better to go with something like (not tested!)
```ruby
if RUBY_VERSION > '2.7'
  inspect = ::Kernel.method(:inspect)
  define_method(:inspect) do
    orig = inspect.bind_call(self)
    "#{orig[0...-1]}#{__inspect}>"
  end
else
  def inspect
    "#<#{self.class}:#{"0x0000%x" % (__id__ << 1)}#{__inspect}>"
  end
end
```

Reply:
@flash-gordon fixed by https://github.com/dry-rb/dry-core/pull/53/commits/6c8a6fa9141f5297d3b602ad35644fe20b8417fe